specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,282 @@
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/{schemaVersion}?api-version={apiVersion}
+//  POST https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/:get-id?api-version={apiVersion}
+//  PUT https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}?api-version={apiVersion}
+@server(
+  "{fullyQualifiedNamespace}",
+  "The Schema Registry service endpoint.",
+  {
+    @doc("The Schema Registry service endpoint, for example 'my-namespace.servicebus.windows.net'.")
+    fullyQualifiedNamespace: url,
+  }
+)
+@versioned(ServiceApiVersions)
+@doc("SchemaRegistryClient is a client for registering and retrieving schemas from the Azure Schema Registry service.")
+namespace Azure.SchemaRegistry;
+
+@doc("Represents the Schema Registry API version to use for requests.")
+enum ServiceApiVersions {
+  @useDependency(Azure.Core.Versions.v1_0_Preview_1)
+  @doc("Azure Schema Registry 2021-10 Version")
+  V2021_10: "2021-10",
+
+  @useDependency(Azure.Core.Versions.v1_0_Preview_1)
+  @doc("Azure Schema Registry 2022-10 Version")
+  V2022_10: "2022-10",
+
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @doc("Azure Schema Registry 2023-07-01 Version. This is the default version.")
+  V2023_07_01: "2023-07-01",
+}
+
+@doc("Azure Active Directory OAuth2 Flow")
+model AzureAuth
+  is OAuth2Auth<[
+    {
+      type: OAuth2FlowType.implicit;
+      authorizationUrl: "https://login.microsoftonline.com/common/oauth2/authorize";
+
+      // add description to below: "impersonate your user account"
+      scopes: ["https://eventhubs.azure.net/.default"];
+    }
+  ]>;
+
+// ROUTES
+alias resourceOperations = Azure.Core.StandardResourceOperations;
+
+interface SchemaOperations {
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups?api-version={apiVersion}
+  @summary("Get list of schema groups.")
+  @doc("Gets the list of schema groups user is authorized to access.")
+  @sharedRoute
+  listSchemaGroups is resourceOperations.ResourceList<SchemaGroup>;
+
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/$schemas/{id}?api-version={apiVersion}
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ SchemaId resource doesn't allow returning binary schema and headers."
+  @summary("Get a registered schema by its unique ID reference.")
+  @doc("Gets a registered schema by its unique ID.  Azure Schema Registry guarantees that ID is unique within a namespace. Operation response type is based on serialization of schema requested.")
+  @route("/$schemaGroups/$schemas/{id}")
+  @get
+  getSchemaById is Azure.Core.Foundations.Operation<
+    Azure.Core.Foundations.ItemKeysOf<SchemaId>,
```

Comment:
I believe so, otherwise, it seems to not unpack the `id` property in SchemaId. `error @typespec/http/missing-path-param: Path contains parameter id but wasn't found in given parameters`